cdisc_rules_engine/utilities/base_report.py

```diff
@@ -0,0 +1,167 @@
+            if (self._item_type == "list")
+            else (x["rule_id"], x["domain"]),
+        )
+
+    def _generate_error_details(self, validation_result) -> List[List]:
+        """
+        Generates the Issue details data that goes into the excel export.
+        Each row is represented by a list or a dict containing the following information:
+        return [
+            "RuleID",
+            "Message",
+            "Severity",
+            "Dataset",
+            "USUBJID",
+            "Record",
+            "Sequence",
+            "Variable(s)",
+            "Value(s)"
+        ]
+        """
+        errors = []
+        for result in validation_result.results or []:
+            if result.get("errors", []) and result.get("executionStatus") == "success":
+                variables = result.get("variables", [])
+                for error in result.get("errors"):
+                    error_item = {
+                        "rule_id": validation_result.id,
+                        "message": result.get("message"),
+                        "severity": validation_result.severity,
+                        "domain": result.get("domain"),
+                        "uSubjId": error.get("uSubjId", ""),
+                        "row": error.get("row", ""),
+                        "seq": error.get("seq", ""),
+                    }
+
+                    if self._item_type == "list":
+                        error_item["variables"] = ", ".join(variables)
+                        error_item["values"] = ", ".join(
+                            [
+                                str(error.get("value", {}).get(variable))
+                                for variable in variables
+                            ]
+                        )
+                        errors = errors + [[*error_item.values()]]
+                    elif self._item_type == "dict":
+                        error_item["variables"] = variables
+                        error_item["values"] = [
+                            str(error.get("value", {}).get(variable))
+                            for variable in variables
+                        ]
+                        errors = errors + [error_item]
+        return errors
+
+    def get_rules_report_data(self) -> List[List]:
+        """
+        Generates the rules report data that goes into the excel export.
+        Each row is represented by a list or a dict containing the following information:
+        [
+            "RuleID",
+            "Version",
+            "Message",
+            "Status"
+        ]
+        """
+        rules_report = []
+        for validation_result in self._results:
+            rules_item = {
+                "rule_id": validation_result.id,
+                "version": "1",
+                "message": validation_result.message,
+                "status": "SUCCESS"
+                if validation_result.execution_status == "success"
+                else "SKIPPED",
+            }
```

Comment:
`Please` use ExecutionStatus enum on lines 148, 9

Reply:
Updated it for lines 147, 148, 149.